daemon/cmd/daemon_main.go

```diff
@@ -1850,22 +1850,7 @@ func startDaemon(d *Daemon, restoredEndpoints *endpointRestoreState, cleaner *da
 		})
 		ms.CollectStaleMapGarbage()
 		ms.RemoveDisabledMaps()
-
-		if len(d.restoredCIDRs) > 0 {
-			// Release restored CIDR identities after a grace period (default 10
-			// minutes).  Any identities actually in use will still exist after
-			// this.
-			//
-			// This grace period is needed when running on an external workload
-			// where policy synchronization is not done via k8s. Also in k8s
-			// case it is prudent to allow concurrent endpoint regenerations to
-			// (re-)allocate the restored identities before we release them.
-			time.Sleep(option.Config.IdentityRestoreGracePeriod)
-			log.Debugf("Releasing reference counts for %d restored CIDR identities", len(d.restoredCIDRs))
-			d.ipcache.ReleaseCIDRIdentitiesByCIDR(d.restoredCIDRs)
-			// release the memory held by restored CIDRs
-			d.restoredCIDRs = nil
-		}
+		go d.releaseRestoredCIDRs() // called from goroutine as it sleeps
```

Comment:
This is already in a goroutine from line 1842, I don't see the need to spawn an additional one here. If we want to make the sleep more obvious, I would keep it in this function and leave `releaseRestoredCIDRs` to just the business logic of actually releasing the CIDRs. But that's more a matter of preference, I don't feel strongly either way about that.
```suggestion
		d.releaseRestoredCIDRs()
```

Reply:
OK, fair enough. I'll just add a comment of ("no code after this line, it sleeps for an arbitrary time")